packages/dashboard/pages/playground/[functionId].tsx

```diff
@@ -46,21 +63,60 @@ const PlaygroundPage = () => {
               code = monaco.editor.getModels()[1].getValue();
             }
 
+            let tsCode = '';
+
             setIsLoading(true);
 
+            if (esbuildStatus === ESBuildStatus.Success) {
+              try {
+                tsCode = code;
+                const files = new Map<string, { content: string }>();
+
+                files.set('index.ts', {
+                  content: code,
+                });
+
+                const esbuildResult = await build(files);
+
+                if (esbuildResult.outputFiles?.[0]) {
+                  code = esbuildResult.outputFiles[0].text;
+                } else {
+                  esbuildResult.errors.forEach(e => {
+                    console.error(e);
+                  });
+                }
+              } catch (e) {
+                console.error(e);
+              }
+            }
+
             const deployment = await createDeployment.mutateAsync({
               functionId: func.id,
               functionSize: new TextEncoder().encode(code).length,
+              tsSize: new TextEncoder().encode(tsCode).length,
               assets: [],
             });
 
-            await fetch(deployment.codeUrl, {
-              method: 'PUT',
-              headers: {
-                'Content-Type': 'text/javascript',
-              },
-              body: code,
-            });
+            await Promise.all([
+              fetch(deployment.codeUrl, {
+                method: 'PUT',
+                headers: {
+                  'Content-Type': 'text/javascript',
+                },
+                body: code,
+              }),
+              ...(tsCode.length > 0
+                ? [
+                    fetch(deployment.tsCodeUrl!, {
```

Comment:
I'm wondering if it makes sense to only upload the TS code when we previously had a TS code? Meaning TS is only supported when creating Functions via the dashboard, and not when creating them via the CLI (which makes sense because we can't support this)

Reply:
I understand what you're saying. Firstly, V8 can only run JS code, which means we only have two options: one is to store only TS code and transpile it to JS at runtime, but it will reduce our performance, so I don't recommend this approach. The other option is to store both TS and JS code, but it will cost us more space. I chose the latter because for users, response time is the most important factor. Also, as you mentioned, the CLI results will only support JS, and I think this is unavoidable in the current scenario. If you want to avoid this problem, the playground would have to implement a file system (such as using a webcontainer), which would involve a lot of work and is not within the scope of this PR.

Reply:
Completely agree. We could probably add a new column to the `Function` table that will be used to check if the function has been created using the CLI or the Dashboard. With that, we could easily only show the Playground button if the Function has been created from the Dashboard.

Reply:
Give it a try and see if it meets your requirements.

Reply:
Thanks, I'll take a look this weekend and we then should be ready to merge

Reply:
@QuiiBz How is the progress, can this pr be merged?